src/__tests__/parsers/index.test.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+
+jest.mock('../../parsers/babylon_parser', () => {
+  const mock_js_parse = jest.fn();
+  return { parse: mock_js_parse, mock_js_parse };
+})
+jest.mock('../../parsers/typescript_parser', () => {
+  const mock_ts_parse = jest.fn();
+  return { parse: mock_ts_parse, mock_ts_parse };
+})
+
+import {parse} from '../../parsers';
+import { mock_js_parse } from '../../parsers/babylon_parser';
+import { mock_ts_parse } from '../../parsers/typescript_parser';
+
+describe('select parser', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  });
+  it('for .js or .jsx file', () => {
+    const files = ["abc.js", "abc.jsx"]
+    files.forEach(file => {
+      parse(file, undefined, true);
+      expect(mock_js_parse).toHaveBeenCalled();
```

Comment:
the clearAllMocks already happened in the outer describe's beforeEach()...

Reply:
But in my opinion that is called only before each `it` block, not before each single `file`.

Reply:
perhaps this helps: https://jestjs.io/docs/en/setup-teardown#scoping

Reply:
But `beforeEach` nevertheless gets called in the following sample only once:

```js
describe('test', () => {
  beforeEach(() => {
    // executed only once
    console.log('beforeEach');
  });
  it('test', () => {
    [1, 2, 3].forEach(v => {
      // called three times
      console.log(v);
    });
  });
});
```

In the case of the test we are talking about that means, that it falsely succeeds in the case that:

* `mock_js_parse`, but not `mock_ts_parse` gets being called within parse for the first entry ("abc.js")
* and for the second entry ("abc.jsx") neither `mock_js_parse` nor `mock_ts_parse` gets called

since the mock functions only get reset before each `it` block, but not before each call of the `forEach` callback.
Or am I completely wrong?

Reply:
ok, I am an idiot, I misunderstood, but now I see what you are talking about, you are absolutely right.

Reply:
It's fine, I also could have formulated it better in the first place.